src/System.Runtime.Extensions/tests/System/IO/PathTestsBase.cs

```diff
@@ -201,6 +201,236 @@ public partial class PathTestsBase
             { @"C:\\foo2", @"C:\" },
         };
 
+        public static TheoryData<string, string, string> GetFullPath_Windows_NonFullyQualified()
+        {
+            string curDir = @"C:\git\corefx";
+            string root = @"C:\";
+
+            return new TheoryData<string, string, string>
+            {
+                { curDir, curDir, curDir },
+                { "", curDir, curDir },
+                { "..", curDir, Path.GetDirectoryName(curDir) },
+                { curDir + @".\.\.\.\.\.", curDir, curDir },
+                { curDir + new string(Path.DirectorySeparatorChar, 3) + ".", curDir, curDir },
+                { curDir + @"\..\" + Path.GetFileName(curDir) + @"\.\..\" + Path.GetFileName(curDir), curDir, curDir },
+                { root + @"\somedir\..", curDir, root },
+                { root, curDir, root },
+                { root + @"..\..\..\..", curDir, root },
+                { root + new string(Path.DirectorySeparatorChar, 3), curDir, @"C:\" },
+
+                // Current drive rooted
+                { @"\tmp\bar", curDir, root + @"tmp\bar" },
+                { @"\.\bar", curDir, root + @"bar" },
+                { @"\tmp\..", curDir, Path.GetPathRoot(curDir).Substring(0, 3) },
+                { @"\tmp\bar\..", curDir, root + @"tmp" },
+                { @"\tmp\bar\..", curDir, root + @"tmp" },
+                { @"\", curDir, root },
+
+                // Specific drive rooted
+                { root[0] + @":tmp\foo\..", curDir, curDir + @"\tmp" },
+                { root[0] + @":tmp\foo\.", curDir, curDir + @"\tmp\foo" },
+                { root[0] + @":tmp\foo\..", curDir, curDir + @"\tmp" },
+                { root[0] + @":tmp", curDir, curDir + @"\tmp" },
+                { root[0] + @":", curDir, curDir },
+                { root[0].ToString(), curDir, curDir + Path.AltDirectorySeparatorChar + root[0] },
+
+                { @"Z:tmp\foo\..", curDir, @"Z:\tmp" },
+                { @"Z:tmp\foo\.", curDir, @"Z:\tmp\foo" },
+                { @"Z:tmp\foo\..", curDir, @"Z:\tmp" },
+                { @"Z:tmp", curDir, @"Z:\tmp" },
+                { @"Z:", curDir, @"Z:\" },
+                { @"Z", curDir, curDir + Path.DirectorySeparatorChar + @"Z" }
+            };
+        }
+
+        public static TheoryData<string, string, string> GetFullPath_Windows_UNC()
+        {
+            string root = @"\\?\";
+            var data = new TheoryData<string, string, string>();
+
+            data.Add(@"foo", root + @"UNC\", root + @"UNC\foo");
```

Comment:
Where is `\\?\` as a root identical to `\\.\` and where do they differ? I can't easily tell from looking at this. It would be much easier to understand what was going on here if the identical cases were all in one set of input data. Additionally it is highly valuable to easily tell what cases are considered to be identical and what are different. My first knee jerk expectation is that `\\?\` and `\\.\` always generate identical results. Is that true? How do I tell by looking at this? Also, if it is true, someone could introduce a test case here that negates our assertion without it being noticed that our assertion doesn't hold.

I can easily audit what is going on when I see full strings like. It is way harder for me to check what you've done when you're building these up when you don't technically have to.

``` C#
  // I have to keep processing in my head what root is
  data.Add(@"foo", root + @"UNC\", root + @"UNC\foo");
 
  // It's all here, I don't need to keep looking for root
  { @"foo", "\\?\UNC\", "\\?\UNC\foo" }
```
You can pipe the input through another helper that turns all `\\?\` into `\\.\` and I can easily understand what you're asserting.

Reply:
(Note that I know we don't get the same results for all cases with `\\?\` vs. `\\.\`, but currently this doesn't capture why or where they differ.)